plugins/modules/package_repository.py

```diff
@@ -0,0 +1,295 @@
+    if module.params["disable_sources"] is not None:
+        data["disable_sources"] = module.params["disable_sources"]
+    if module.params["components"]:
+        data["components"] = ",".join(module.params["components"])
+    if module.params["arches"]:
+        data["arches"] = ",".join(module.params["arches"])
+    if module.params["key"]:
+        data["key"] = module.params["key"]
+    if module.params["enabled"] is not None:
+        data["enabled"] = module.params["enabled"]
+
+    return data
+
+
+def create_package_repository(module, client: Client):
+    data = data_for_create_package_repository(module)
+    package_repository = PackageRepository.create(client, data)
+    if module.params["disable_sources"]:  # this parameter can only be set with put
+        data = data_for_update_package_repository(module, package_repository)
+        package_repository.update(client, data)
+    return (
+        True,
+        package_repository.to_ansible(),
+        dict(before={}, after=package_repository.to_ansible()),
+    )
+
+
+def data_for_update_package_repository(module, package_repository):
+    data = {}
+    if module.params["name"]:
+        if package_repository.name != module.params["name"]:
+            data["name"] = module.params["name"]
+    if module.params["url"]:
+        if package_repository.url != module.params["url"]:
+            data["url"] = module.params["url"]
+    if module.params["distributions"]:
+        if package_repository.distributions != module.params["distributions"]:
+            data["distributions"] = ",".join(module.params["distributions"])
+    if module.params["disabled_pockets"]:
+        if package_repository.disabled_pockets != module.params["disabled_pockets"]:
+            data["disabled_pockets"] = ",".join(module.params["disabled_pockets"])
+    if module.params["disabled_components"]:
+        if (
+            package_repository.disabled_components
+            != module.params["disabled_components"]
+        ):
+            data["disabled_components"] = ",".join(module.params["disabled_components"])
+    if module.params["disable_sources"] is not None:
+        if package_repository.disable_sources != module.params["disable_sources"]:
+            data["disable_sources"] = module.params["disable_sources"]
+    if module.params["components"]:
+        if package_repository.components != module.params["components"]:
+            data["components"] = ",".join(module.params["components"])
+    if module.params["arches"]:
+        if package_repository.arches != module.params["arches"]:
+            data["arches"] = ",".join(module.params["arches"])
+    if module.params["key"]:
+        if package_repository.key != module.params["key"].rstrip("\n"):
+            data["key"] = module.params["key"]
+    if module.params["enabled"] is not None:
+        if package_repository.enabled != module.params["enabled"]:
+            data["enabled"] = module.params["enabled"]
+    return data
```

Comment:
Not especially important, but this looks very similar to `data_for_create_package_repository` with the extra `if <changed>` check. Is it worth condensing down into a single function that can be called for both?